tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java

```diff
@@ -43,25 +39,26 @@ protected SetType(TiColumnInfo.InternalTypeHolder holder) {
   /** {@inheritDoc} */
   @Override
   protected Object decodeNotNull(int flag, CodecDataInput cdi) {
-    throw new UnsupportedTypeException("Set type not supported");
+    if (flag != Codec.UVARINT_FLAG) throw new TypeException("Invalid IntegerType flag: " + flag);
+    return this.getElems().get((int) IntegerCodec.readUVarLong(cdi) - 2);
   }
 
   /** {@inheritDoc} */
   @Override
   protected void encodeKey(CodecDataOutput cdo, Object value) {
-    throw new UnsupportedTypeException("Set type not supported");
+    IntegerCodec.writeULongFully(cdo, Converter.convertToLong(value), true);
```

Comment:
How would you test write encode process? I suppose here you might need to add index and filter pushdown tests for these two types.

Reply:
@ilovesoup set or enum cannot be pushed down to tikv. So encoding logic is not necessary. We can do it later once tikv allows set or enum to be pushed down to tikv.